tests/expectations/metrics/column_map_metrics/test_unexpected_indices_and_query.py

```diff
@@ -305,7 +310,159 @@ def test_sa_unexpected_index_query_metric_without_id_pk(sa, animal_table_df):
         "unexpected_condition": unexpected_columns_metric,
         "table.columns": table_columns_metric,
     }
-    with pytest.raises(MetricResolutionError):
-        results: Dict[Tuple[str, str, str], MetricValue] = engine.resolve_metrics(
-            metrics_to_resolve=(unexpected_index_query,), metrics=metrics
+    results: Dict[Tuple[str, str, str], MetricValue] = engine.resolve_metrics(
+        metrics_to_resolve=(unexpected_index_query,), metrics=metrics
+    )
+    for (key, val) in results.items():
+        assert (
+            val == "SELECT animals \n"
+            "FROM test \n"
+            "WHERE animals IS NOT NULL AND (animals NOT IN ('cat', 'fish', 'dog'));"
+        )
+
+
+@pytest.mark.unit
+def test_spark_unexpected_index_list_metric_with_id_pk(
+    spark_session, animal_table_df, metric_value_kwargs_complete
+):
+    engine: SparkDFExecutionEngine = build_spark_engine(
+        spark=spark_session, df=animal_table_df, batch_id="my_id"
+    )
+
+    (
+        table_columns_metric,
+        unexpected_columns_metric,
+        metrics,
+    ) = _build_table_columns_and_unexpected(engine, metric_value_kwargs_complete)
+
+    unexpected_index_list = MetricConfiguration(
+        metric_name=f"column_values.in_set.{SummarizationMetricNameSuffixes.UNEXPECTED_INDEX_LIST.value}",
+        metric_domain_kwargs={"column": "animals"},
+        metric_value_kwargs=metric_value_kwargs_complete,
+    )
+    unexpected_index_list.metric_dependencies = {
+        "unexpected_condition": unexpected_columns_metric,
+        "table.columns": table_columns_metric,
+    }
+    results: Dict[Tuple[str, str, str], MetricValue] = engine.resolve_metrics(
+        metrics_to_resolve=(unexpected_index_list,), metrics=metrics
+    )
+    for (key, val) in results.items():
+        assert val == [
+            {"animals": "giraffe", "pk_1": 3},
+            {"animals": "lion", "pk_1": 4},
+            {"animals": "zebra", "pk_1": 5},
+        ]
+
+
+@pytest.mark.unit
+def test_spark_unexpected_index_list_metric_without_id_pk(
+    spark_session, animal_table_df
```

Comment:
Same note here and throughout regarding spark session

Reply:
Done